crypto/rand/rand_unix.c

```diff
@@ -172,29 +168,43 @@ int RAND_poll(void)
                 continue;
             setbuf(fp, NULL);
             if (fread(temp, 1, TEMPSIZE, fp) == TEMPSIZE) {
-                RAND_add(temp, TEMPSIZE, TEMPSIZE);
-                ok++;
+                cb(arg, temp, TEMPSIZE, TEMPSIZE);
                 fclose(fp);
-                break;
+                goto done;
             }
+            fclose(fp);
         }
     }
 #   endif
 
-#   ifdef OPENSSL_RAND_SEED_GETRANDOM
+#   ifdef OPENSSL_RAND_SEED_RDTSC
```

Comment:
Suggest moving RDTSC and RDCPU up above GETRANDOM.  RDTSC isn't producing much but is very fast and might as well be included.  RDCPU is very fast and would be preferable to `getrandom(2)` if both are enabled.